src/silx/io/specfile/src/sfmca.c

```diff
@@ -298,6 +298,11 @@ SfGetMca( SpecFile *sf, long index, long number, double **retdata, int *error )
        val = PyMcaAtof(strval);
        data[vals] = val;
        vals++;
+     } else if (i>0) {
+         strval[i] = '\0';
+         val = PyMcaAtof(strval);
+         data[vals] = val;
```

Comment:
This could potentially lead to a buffer overflow if the mca has 2049 entries. (Same as the block at line 295)

Reply:
The data pointer seems to be resized when it becomes larger than initsize(2048).

https://github.com/silx-kit/silx/blob/8f146767a22bb100f8ce6adf73377fea91b899ff/src/silx/io/specfile/src/sfmca.c#L269-L282

In my case, there was no buffer overflow even at 2560 entries.

Reply:
Exactly my point. The code that resizes the buffer is not reached when the mca has 2049 entries.